### ISSUE TITLE
useTrackedState does not work like before and always cause re-render for any state update

### ISSUE THREAD

**Issue by ulion** [dai-shi/react-tracked#147]
I think this was introduced by:
https://github.com/dai-shi/react-tracked/blob/51d7fee95cdd2b15d322f114f83ecf773c3e3143/src/createContainer.ts#L80

before this commit, useTrackedState was memorized, but in this commit, it became directly calling underline createTrackedSelector(useSelector)

If I rollback to 1.7.4, it works like before and as expected.

**Comment by dai-shi** [comment]
Thanks for reporting.
That was quite a bit of refactoring, but I'm not sure how it causes a problem.
Would you be able to create a reproduction with codesandbox?

**Comment by dai-shi** [comment]
So, is that mean you use createContainer inside render function? That useMemo usage in the previous version is actually wrong and but maybe it worked for you by accident. Without your repro, can’t say for sure.

**Comment by ulion** [comment]
No, the createContainer is called only once, in the global scope.

```
export const {
  Provider: ComposingStateProvider,
  useTrackedState: useComposingState,
  useUpdate: useUpdateComposingState,
} = createContainer(() => useState({
  composing: false,
//  compositionText: '',
}))
```

My previous reply about the WeakMap maybe incorrect, but without memo, now each time run useTrackedState, it run a different new function (though same function body), and before it run the exact same function.
Still no idea why this caused any update in state will cause all places be re-rendered yet.

**Comment by ulion** [comment]
update: 1.7.5 is fine, 1.7.8 is not. Testing other versions...

**Comment by dai-shi** [comment]
The change we made in v1.7.7 is #139.
Yeah, please keep investigating. Would be nice to have a minimal reproduction.

**Comment by ulion** [comment]
the issue was introduced in commit a211d93568bbebc5d4cb235f7e49b6a83014290e

**Comment by dai-shi** [comment]
Good catch. Can you reproduce the issue with codesandbox?
(So, you get `useContext(StateContext)===null` but it's not what you expect? Anyway, a minimal reproduction should help.)

**Comment by ulion** [comment]
I think clearly in development mode, useContext(StateContext) call added here will cause it will be re-rendered anyway, that's the problem.

**Comment by dai-shi** [comment]
Oh yeah, the issue is just about extra re-renders. Got it. Curious why our tests didn't catch it.

**Comment by dai-shi** [comment]
Please try: https://ci.codesandbox.io/status/dai-shi/react-tracked/pr/148
Find "Local Install Instructions" ☝️

**Comment by ulion** [comment]
I can confirm that PR did fix the issue in development mode. great!

**Comment by dai-shi** [comment]
Thanks for confirming! and reporting too. I totally missed it. Your patience is appreciated.

**Comment by dai-shi** [comment]
Published: https://www.npmjs.com/package/react-tracked/v/1.7.9
